### PR TITLE
feat: Implement new link field

### LINKED ISSUE #5116: Wizard Step 2: Cannot change first option of social links (website)
The social links section should 
* allow the first link to change the option from website to something else, e.g. Twitter.
* have the option "Event Site" in the dropdown instead of "External Event URL"

There is no need to have two + buttons.
* A plus button can be anything website or social link. There is no need to distinguish between those.

Also there is an issue if a user chooses two websites: 
* When the user saves the choice and then returns to the social links in the wizard, the website appears as an editable field. 
* Expected: It should be a dropdown item as before.

Screenshot from 2020-09-21 19-18-33

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #5116 

image

### CODE REVIEW

File: app/components/widgets/forms/link-field.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  get fixedValue() {
+    return this.fixValue(this.value);
+  }
+
+  fixValue(value: string): string {
+    const splitted = value.split(this.prefix);
+    return splitted[1] || splitted[0];
+  }
+
+  @action
```
Comment (niranjan94) [inline app/components/widgets/forms/link-field.ts:43]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6395020)

File: app/components/widgets/forms/link-field.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  setValue(event: any) {
```
Comment (niranjan94) [inline app/components/widgets/forms/link-field.ts:44]: Codacy found an issue: [Argument 'event' should be typed with a non-any type.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6395020)

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- app/components/widgets/forms/link-field.ts  2
- tests/integration/components/widgets/forms/link-field-test.ts  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6395020&bid=20738634)

File: app/components/widgets/forms/link-field.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  get prefix(): string {
+    return this.args.prefix ?? 'https://';
+  }
+
+  get finalValue(): string {
+    return this.prefix + this.fixedValue;
+  }
+
+  get fixedValue() {
```
Comment (niranjan94) [inline app/components/widgets/forms/link-field.ts:34]: Codacy found an issue: [Missing return type on function.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6395020)

**Comment by mariobehling** [comment]
Nice!

* Please change
Flicker -> Flickr https://flickr.com

* Please add
  * Telegram https://t.me/
  * Gitter https://gitter.im/

**Comment by iamareebjamal** [comment]
Done

**Pull request merged**